src/Uno.Extensions.Core.Generators/PropertySelector/PropertySelectorCandidate.cs

```diff
@@ -1,43 +1,201 @@
-﻿using System;
+﻿using System.Collections.Generic;
+using System.Collections.Immutable;
 using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
 using System.Linq;
+using System.Runtime.CompilerServices;
 using System.Threading;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.CSharp;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Uno.Extensions.Core.Generators.Helpers;
+using Uno.Extensions.Edition;
 
 namespace Uno.Extensions.Generators.PropertySelector;
 
 internal readonly record struct PropertySelectorCandidate
 {
 	public PropertySelectorCandidate(GeneratorSyntaxContext context, CancellationToken ct)
 	{
-		var syntax = context.Node as InvocationExpressionSyntax;
-		var position = syntax is null ? default : syntax.SyntaxTree.GetLineSpan(syntax.Span);
-		var method = syntax is null
-			? null
-			: context.SemanticModel.GetSymbolInfo(syntax, ct).Symbol as IMethodSymbol;
-		var hasPropertySelector = method?.Parameters.Any(p => p.Type is INamedTypeSymbol
-		{
-			IsGenericType: true,
-			MetadataName: "PropertySelector`2",
-			ContainingNamespace: { Name: "Edition", ContainingNamespace: { Name: "Extensions", ContainingNamespace.Name: "Uno" } }
-		});
-
-		Context = context;
-		Syntax = syntax;
-		Location = position;
-		Method = method;
-		IsValid = hasPropertySelector ?? false;
-	}
+		Syntax = (InvocationExpressionSyntax)context.Node;
+		Location = Syntax.SyntaxTree.GetLineSpan(Syntax.Span);
+		
+		var method = context.SemanticModel.GetSymbolInfo(Syntax, ct).Symbol as IMethodSymbol;
+		
+		if (method is null ||
+			!method.Parameters.Any(p => p.Type is INamedTypeSymbol
+			{
+				IsGenericType: true,
+				MetadataName: "PropertySelector`2",
+				ContainingNamespace: { Name: "Edition", ContainingNamespace: { Name: "Extensions", ContainingNamespace.Name: "Uno" } }
+			}) ||
+			method.Parameters.FirstOrDefault(p => p.HasAttribute<CallerFilePathAttribute>()) is not { } callerPathParameter ||
+			method.Parameters.FirstOrDefault(p => p.HasAttribute<CallerLineNumberAttribute>()) is not { } callerLineParameter
+			)
```

Comment:
Logic-wise they are close, but analyzer implementation relies more on symbols.

Reply:
Then could we extract them in an helper? The idea I that I've the feeling we could easily update one and forget to update the other one as they are in 2 different classes and not even sharing the same name.

But I may be wrong: the 2 classes are searching for the same nodes but in a different way. I didn't validated but maybe there are plenty of other places in those classes that should be shared to avoid such issue, so it's irrelevant to do it only for this part? Your call :)